gammapy/data/utils.py

```diff
@@ -151,3 +159,39 @@ def get_irfs_features(
         features.meta["frame"] = "galactic"
 
     return features
+
+
+def check_time_intervals(time_intervals):
+    """Function checking that the intervals are made of `astropy.time.Time` objects and are disjoint.
+
+    Parameters
+    ----------
+    time_intervals : array of intervals of `astropy.time.Time`, an interval being a list
+
+    Returns
+    -------
+    valid: bool
+    """
+    # Note: this function will be moved into a future class `TimeInterval`
+    ti = np.asarray(time_intervals)
+    if ti.shape == () or ti.shape == (0,):
+        return False
+
+    if ti.shape == (2,) or ti.shape == (2, 1):
+        if not np.all([isinstance(_, Time) for _ in ti]):
+            return False
+    else:
+        for xx in ti:
+            if not np.all([isinstance(_, Time) for _ in xx]):
+                return False
+        if not CHECK_OVERLAPPING_TI:
+            return True
+        for xx in combinations(ti, 2):
+            i1 = [xx[0][0].to_value("gps"), xx[0][1].to_value("gps")]
+            i2 = [xx[1][0].to_value("gps"), xx[1][1].to_value("gps")]
+            if interval_overlap_length(i1, i2) > 0.0:
```

Comment:
this might trigger precision error, maybe it explains why it fails for the HAWC dataset.

Reply:
No, for HAWC there are really overlapping GTI